content/using_ec2_spot_instances_with_eks/spotmanagednodegroups/workers_eksctl.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Adding Spot Workers with eksctl"
+date: 2018-08-07T11:05:19-07:00
+weight: 30
+draft: false
+---
+
+In this section we will deploy the instance types we selected and request nodegroups that adhere to Spot diversification best practices. For that we will use **[eksctl create nodegroup](https://eksctl.io/usage/managing-nodegroups/)** to generate the ClusterConfig file and save the output. The ClusterConfig file will be edited with additional configuration including filtering the instance types, and use the edited eksctl configuration files to add the new nodes to the cluster.
+
+Let's first create the configuration file:
+
+```bash
+eksctl create nodegroup \
+    --cluster=eksworkshop-eksctl \
+    --region=$AWS_REGION \
+    --dry-run \
+    --managed \
+    --spot \
+    --name=dev-4vcpu-16gb-spot \
+    --instance-selector-vcpus=4 \
+    --instance-selector-memory=16 \
+    --instance-selector-cpu-architecture=x86_64 \
+    --instance-selector-gpus=0 \
+    > ~/environment/spot_nodegroup_4vcpu_16gb.yml
+
+eksctl create nodegroup \
+    --cluster=eksworkshop-eksctl \
+    --region=$AWS_REGION \
+    --dry-run \
+    --managed \
+    --spot \
+    --name=dev-8vcpu-32gb-spot \
+    --instance-selector-vcpus=8 \
+    --instance-selector-memory=32 \
+    --instance-selector-cpu-architecture=x86_64 \
+    --instance-selector-gpus=0 \
+    > ~/environment/spot_nodegroup_8vcpu_32gb.yml
+```
+
+This will create 2 files, `spot_nodegroups_4vcpu_16gb.yml` and `spot_nodegroups_8vcpu_32gb.yml`, that we will use to instruct eksctl to create two nodegroups, both with a diversified configuration.
+
+Let's edit the 2 configuration files before using it to create the node groups:
+
+1. Change the **maxSize** to 5 and **minSize** to 1. Leave **desiredCapacity** at 2. 
+
+1. Delete the *d3en*, *h1*, *m5zn* and *g4dn* instances from the **instanceTypes**. Leave in only the Instances from M and T families (except m5zn.large and m5zn.xlarge).
```

Comment:
this approach needs to be reconsidered. I don't like this step and if we can't get selector to create a proper list then we just shouldn't use it

Reply:
I have removed the use of EKSCTL instance selector and added instance selector CLI.